plugins/jobs/src/js/JobsOverview.tsx

```diff
@@ -0,0 +1,165 @@
+import * as React from "react";
+import Router, {
+  InjectedRouter,
+  LocationDescriptor,
+  Link,
+  withRouter
+} from "react-router";
+
+// tslint:disable-next-line:no-submodule-imports
+import { BehaviorSubject } from "rxjs/BehaviorSubject";
+// tslint:disable-next-line:no-submodule-imports
+import "rxjs/add/observable/empty";
+
+import { graphqlObservable, componentFromStream } from "data-service";
+import gql from "graphql-tag";
+
+// import JobsTabLoading from "./components/JobsTabLoading";
+// import JobsTabError from "./components/JobsTabError";
+// import JobsTabList from "./components/JobsTabList";
+// import JobsTabEmpty from "./components/JobsTabEmpty";
+
+import JobModel from "./data/JobModel";
+import { combineLatest } from "rxjs/observable/combineLatest";
+import { Observable, Subscribable } from "rxjs/Observable";
+import { JobConnection } from "#PLUGINS/jobs/src/js/types/JobConnection";
+import { Job } from "#PLUGINS/jobs/src/js/types/Job";
+
+interface JobsOverviewProps {
+  path: string[];
+  router: InjectedRouter;
+  location: LocationDescriptor;
+  params: Router.Params;
+}
+
+// const withRouter = getContext({
+//   router: routerShape
+// });
+
+function DummyComponent(props: any): JSX.Element {
+  function updateFilter(event: React.FormEvent<HTMLInputElement>) {
+    props.handleFilterChange(event.currentTarget.value);
+  }
+
+  console.log("rendering with props", props, props.pathname);
+  return (
+    <div style={{ background: "white" }}>
+      <p>Path</p>
+      <p>{props.data && props.data.path.join("/")}</p>
+      <p>Filter</p>
+      <input value={props.filter} onChange={updateFilter} />
+      <p>Jobs</p>
+      {props.data &&
+        props.data.nodes.map((job: Job) => (
+          <p key={job.id}>
+            <Link
+              to={`/jobs/overview2/${props.data.path
+                .concat([job.name])
+                .join(".")}`}
+            >
+              {`link: ${job.name}`}
+            </Link>
+          </p>
+        ))}
+      <p>Children</p>
+      <div>{props.children}</div>
+    </div>
+  );
+}
+
+const JobsOverview = withRouter(
+  componentFromStream<JobsOverviewProps>(
+    (
+      props$: Subscribable<JobsOverviewProps>
+    ): Subscribable<React.ReactNode> => {
+      const jobsOverviewQuery: any = gql`
```

Comment:
Curious why have you went with graphql here?

Reply:
Our understanding when we started working on this whole thing was that we want to get rid of the Structs _now_. This very query (together with the one from Detail Mediator) was the reason why we introduced `JobModel` 🤷‍♂️

Reply:
Yeah, but as I understand this query has nothing to do with a struct it's just to pass the routing params, no?

Reply:
Well, it has nothing to do with old Job Struct, because it replaces it 😅Not sure if I got your question 100%, so let me try to explain it in a bit more detail.

---

When the query is submitted, the `jobs` resolver returns a response with type `JobConnection`, the requested `nodes` field below contains an array of `Job`, which is the new type we introduced with `JobModel` to replace our Structs. 

`JobConnection` Type: https://github.com/dcos/dcos-ui/blob/6880237bdf904d68bb6f554cc7861d01073e2020/plugins/jobs/src/js/types/JobConnection.ts#L6
New `Job` Type used in `JobConnection.nodes`: https://github.com/dcos/dcos-ui/blob/6880237bdf904d68bb6f554cc7861d01073e2020/plugins/jobs/src/js/types/Job.ts#L41 
=> If you check out field resolvers n both files, they contain all the logic Job Struct had before.

--- 

Also filtering logic moved from UI component into `JobModel`, which is the reason why we "just" pass down the routing params 😅
https://github.com/dcos/dcos-ui/blob/6880237bdf904d68bb6f554cc7861d01073e2020/plugins/jobs/src/js/types/JobConnection.ts#L39
https://github.com/dcos/dcos-ui/blob/6880237bdf904d68bb6f554cc7861d01073e2020/plugins/jobs/src/js/types/JobConnection.ts#L44

---

that said, yes, the query is "just" passing down some params to the resolver 😇
If this didnt answer your initial question/concern, can you please try to clarify it?

Reply:
oh, now I see. My initial understanding was that we put `router.push` into graphQL for no real reason

Reply:
Oh, no 😅